daffodil-test/src/test/resources/org/apache/daffodil/extensions/enum/enums.tdml

```diff
@@ -134,7 +134,7 @@
     </element>
 
     <simpleType name="myBit" dfdl:lengthKind="explicit" dfdl:length="1">
-      <restriction base="xs:byte"/>
+      <restriction base="xs:unsignedByte"/>
```

Comment:
I'm concerned we will have a lot of regressions due to this change, and wonder if we need a process of deprecating this via tunable. It might be worth running this change with the regression suite and seeing how many things rely on this behavior.

Reply:
Pretty sure we're going to need the compatibility tunable. 

Particularly in Cyberian applications, it's quite likely that users have schemas with signed integers of length 1 bit, since we've been tolerating that, and their application really doesn't care if the values were 0 and -1 or 0 and 1. 
But their test cases might have expected infoset XML that assumes that 0 and -1 are the values for a signed integer of length 1. 

Our regression test set doesn't include all schemas, and I think we have to assume many others are using DFDL now, and have large sets of schemas. (I recently learned of a group which has 300+ DFDL schemas that we have no access to.)

So  we will need a compatibility tunable like "allowSignedIntegerLength1Bit" or something like that. False means we strictly enforce min length 2 for signed integers. "true" means we issue a warning (which can be suppressed by usual means) but implement the current behavior (which I think creates values 0 and -1. 

Note that issuing a warning, then assuming the user intended to use an unsigned integer instead would be a change in behavior (producing values 0 and 1, not 0 and -1), so we definitely don't want to do that. 

The warning message should suggest converting to use an unsigned integer, and the SDE message if disallowing should also suggest that an unsigned integer is the fix for length 1 bit.

Reply:
Note that there are obscure corner cases involving charset encodings like "bits" where the number of bits could turn out to be 0 or 1 using lengthKind 'pattern'. Detecting these at runtime only is fine with me.  Ex: 
```
<element name="b" type="xs:byte" dfdl:representation="text" dfdl:encoding="bits" dfdl:lengthKind="pattern"
    dfdl:lengthPattern=".{0,1}(?=11111111)" />
```
That captures any zeros and ones followed by a run of 8 consecutive "1"s. 

This is just a corner case, and so long as we don't abort, and if the length turns out to be 0 or 1, we issue an error, then I'm good with it.

Reply:
There were no regressions in our test suite, but Mike makes a great point, so I'll add in the tunable. Do we want to add in a tunable for zero length unsigned integers too i.e allowZeroLengthUnsignedInteger?

Reply:
No. Zero length unsigned integers always fail now (no value), so we don't want to start allowing them since their min length is supposed to be 1 bit.

Reply:
The place where regressions might be found is in running the big regression of 80-some schemas.

Reply:
> The place where regressions might be found is in running the big regression of 80-some schemas.

Right the tests in dfdl-regression-test (which is what I think you're referring to) all passed while using the snapshot version with this change

Reply:
> <element name="b" type="xs:byte" dfdl:representation="text" dfdl:encoding="bits" dfdl:lengthKind="pattern"
>     dfdl:lengthPattern=".{0,1}(?=11111111)" />

I'm not sure I follow. In this case, the code changes wouldn't apply since representation is text, not binary. If we have the following element e1

```
    <xs:element name="e1">
      <xs:complexType>
        <xs:sequence>
          <xs:element name="b" type="xs:byte" dfdl:representation="text" dfdl:encoding="X-DFDL-BITS-LSBF"
                      dfdl:lengthKind="pattern"
                      dfdl:lengthPattern=".{0,1}(?=11111111)" />
          <xs:element name="c" type="xs:string" dfdl:representation="text" dfdl:encoding="X-DFDL-US-ASCII-7-BIT-PACKED"
                      dfdl:lengthKind="explicit" dfdl:length="1" />
        </xs:sequence>
      </xs:complexType>
    </xs:element>
```
and the following data: 011111111, it parses fine and you get

<ex:e1 xmlns:ex="http://example.com"><ex:b>0</ex:b><ex:c></ex:c></ex:e1>

Reply:
> I think you can have `representation="binary"` with `lengthKind="pattern"`.

So we get an SDE if we try that. "Schema Definition Error: Binary data elements cannot have lengthKind='pattern'."

Reply:
I guess I was wrong. Seems like lengthKind="pattern" can't be used to get around the binary length limit.

Reply:
If you use encoding="bits" it is representation="text" You are treating each bit as a letter/character. It's text.